src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/WitnessCommitmentsRule.cs

```diff
@@ -104,37 +112,151 @@ private bool EqualsArray(byte[] a, byte[] b, int length)
         }
 
         /// <summary>
-        /// Gets index of the last coinbase transaction output with SegWit flag.
+        /// Gets commitment in the last coinbase transaction output with SegWit flag.
         /// </summary>
         /// <param name="block">Block which coinbase transaction's outputs will be checked for SegWit flags.</param>
         /// <returns>
-        /// <c>-1</c> if no SegWit flags were found.
-        /// If SegWit flag is found index of the last transaction's output that has SegWit flag is returned.
+        /// <c>null</c> if no SegWit flags were found.
+        /// If SegWit flag is found the commitment of the last transaction's output that has SegWit flag is returned.
         /// </returns>
-        private int GetWitnessCommitmentIndex(Block block)
+        private static Script GetWitnessCommitment(Network network, Block block)
         {
-            int commitpos = -1;
-            for (int i = 0; i < block.Transactions[0].Outputs.Count; i++)
+            Script commitScriptPubKey = null;
+
+            if (network.Consensus.IsProofOfStake && network.Consensus.PosEmptyCoinbase)
             {
-                Script scriptPubKey = block.Transactions[0].Outputs[i].ScriptPubKey;
+                Script scriptPubKey = block.Transactions[0].Inputs[0].ScriptSig;
```

Comment:
Not the best variable name 😂

Reply:
lol... it had a few iterations till I got it right, I will fix.

Reply:
done